pennylane/ops/qubit/observables.py

```diff
@@ -497,6 +505,14 @@ def compute_matrix(basis_state):  # pylint: disable=arguments-differ
          [0. 0. 0. 0.]
          [0. 0. 0. 0.]]
         """
+        if qml.math.get_interface(basis_state) == "jax":
+            basis_state += 0  # convert to int
+            n = len(basis_state)
+            mask = qml.math.flip(2 ** qml.math.arange(n))
+            idx = qml.math.sum(mask * basis_state)
+            mat = qml.math.zeros((2**n, 2**n), like=basis_state)
+            return mat.at[idx, idx].set(1.0)
+
         m = np.zeros((2 ** len(basis_state), 2 ** len(basis_state)))
```

Comment:
```suggestion
        shape = (2 ** len(basis_state), 2 ** len(basis_state))
        if qml.math.get_interface(basis_state) == "jax":
            idx = 0
            for i, m in enumerate(basis_state):
                idx = idx + (m << (len(basis_state) - i - 1))
            mat = qml.math.zeros(shape, like=basis_state)
            return mat.at[idx, idx].set(1.0)

        m = np.zeros(shape)
```

Reply:
For some minimal local testing, these changes seem to work :) I'll commit them and we'll see.